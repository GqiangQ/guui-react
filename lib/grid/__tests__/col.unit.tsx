import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Col from '../col';

configure({adapter: new Adapter()});

describe(Col.displayName, () => {
  it('DisplayName', () => {
    expect(Col.displayName).toEqual('Col');
  });

  describe('className', () => {
    it(' = xxx', () => {
      const tree = renderer
        .create(<Col className="xxx"></Col>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('style', () => {
    it(` = {color: 'red'}`, () => {
      const tree = renderer
        .create(<Col style={{color: 'red'}}></Col>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('span', () => {
    it(` = 2`, () => {
      const tree = renderer
        .create(<Col span={2}></Col>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('offset', () => {
    it(` = 1`, () => {
      const tree = renderer
        .create(<Col offset={1}></Col>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('gutter', () => {
    it(`= 16`, () => {
      const tree = renderer
        .create(<Col gutter={16}></Col>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
