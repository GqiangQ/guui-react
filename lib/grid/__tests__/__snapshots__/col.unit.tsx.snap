// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col className  = xxx 1`] = `
<div
  className="gu-col gu-col-span1"
  style={
    Object {
      "padding": undefined,
    }
  }
>
  <div
    className="xxx"
  />
</div>
`;

exports[`Col gutter = 16 1`] = `
<div
  className="gu-col gu-col-span1"
  style={
    Object {
      "padding": "0 8px",
    }
  }
>
  <div
    className=""
  />
</div>
`;

exports[`Col offset  = 1 1`] = `
<div
  className="gu-col gu-col-span1 gu-col-offset1"
  style={
    Object {
      "padding": undefined,
    }
  }
>
  <div
    className=""
  />
</div>
`;

exports[`Col span  = 2 1`] = `
<div
  className="gu-col gu-col-span2"
  style={
    Object {
      "padding": undefined,
    }
  }
>
  <div
    className=""
  />
</div>
`;

exports[`Col style  = {color: 'red'} 1`] = `
<div
  className="gu-col gu-col-span1"
  style={
    Object {
      "padding": undefined,
    }
  }
>
  <div
    className=""
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;
